yt_dlp/extractor/koo.py

```diff
@@ -0,0 +1,118 @@
+        'info_dict': {
+            'id': 'a2a9c88e-ce4b-4d2d-952f-d06361c5b602',
+            'ext': 'mp4',
+            'title': 'Test for audio',
+            'description': 'md5:ecb9a2b6a5d34b736cecb53788cb11e8',
+            'timestamp': 1632211634,
+            'uploader_id': 'ytdlpTestAccount',
+            'uploader': 'yt-dlpTestAccount',
+            'duration': 214000,
+            'upload_date': '20210921'
+        },
+        'params': {'skip_download': True}
+    }, {  # Test for video
+        'url': 'https://www.kooapp.com/koo/ytdlpTestAccount/a3e56c53-c1ed-4ac9-ac02-ed1630e6b1d1',
+        'info_dict': {
+            'id': 'a3e56c53-c1ed-4ac9-ac02-ed1630e6b1d1',
+            'ext': 'mp4',
+            'title': 'Test for video',
+            'description': 'md5:7afc4eb839074ddeb2beea5dd6fe9500',
+            'timestamp': 1632211468,
+            'uploader_id': 'ytdlpTestAccount',
+            'uploader': 'yt-dlpTestAccount',
+            'duration': 14000,
+            'upload_date': '20210921'
+        },
+        'params': {'skip_download': True}
+    }, {  # Test for link
+        'url': 'https://www.kooapp.com/koo/ytdlpTestAccount/01bf5b94-81a5-4d8e-a387-5f732022e15a',
+        'skip': 'No video/audio found at the provided url.',
+        'info_dict': {
+            'id': '01bf5b94-81a5-4d8e-a387-5f732022e15a',
+            'title': 'Test for link',
+            'ext': 'none',
+        },
+    }, {  # Test for images
+        'url': 'https://www.kooapp.com/koo/ytdlpTestAccount/dc05d9cd-a61d-45fd-bb07-e8019d8ca8cb',
+        'skip': 'No video/audio found at the provided url.',
+        'info_dict': {
+            'id': 'dc05d9cd-a61d-45fd-bb07-e8019d8ca8cb',
+            'title': 'Test for images',
+            'ext': 'none',
+        },
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        data_json = self._download_json(f'https://www.kooapp.com/apiV1/ku/{id}?limit=20&offset=0&showSimilarKoos=true', id)['parentContent']
+        item_json = next(content['items'][0] for content in data_json if try_get(content, lambda x: x['items'][0]['id']) == id)
+        media_json = item_json['mediaMap']
+        formats = []
+
+        mp4_url = media_json.get('videoMp4')
+        video_m3u8_url = media_json.get('videoHls')
+        if mp4_url:
+            formats.append({
+                'url': mp4_url,
+                'ext': 'mp4',
+            })
+        if video_m3u8_url:
+            formats.extend(self._extract_m3u8_formats(video_m3u8_url, id, fatal=False, ext='mp4'))
+        if not formats:
+            raise ExtractorError('No video/audio found at the provided url.', expected=True)
+
+        self._sort_formats(formats)
+        return {
+            'id': id,
+            'title': clean_html(item_json.get('title')),
+            'description': f'{clean_html(item_json.get("title"))}\n\n{clean_html(item_json.get("enTransliteration"))}',
```

Comment:
why is title part of description?

Reply:
This is same as tiktok and twitter tbh. 
There's not description or title per say...there's only post text and post content.
we treat the text as both title and description.
Since title can be trimmed, it's good to have the text in description

Reply:
This is way too complex to be a readable fstring imho. But I'll allow it for now

Reply:
Is this even valid syntax? AFAIK you can't have backslashes in a f-string.

Reply:
you cant have backslashes in the **expresssion part** of fstrings